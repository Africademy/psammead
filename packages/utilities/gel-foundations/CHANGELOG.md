# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.0   | [PR#XX](https://github.com/BBC-News/psammead/pull/XX) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |
