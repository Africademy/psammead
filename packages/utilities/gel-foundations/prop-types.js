module.exports = require('./dist/prop-types'); // eslint-disable-line import/no-unresolved
