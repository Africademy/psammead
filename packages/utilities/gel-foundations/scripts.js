/* eslint-disable global-require */
module.exports = {
  news: require('./dist/scripts/news').default,
  arabic: require('./dist/scripts/arabic').default,
  burmese: require('./dist/scripts/burmese').default,
  cjkThai: require('./dist/scripts/cjk_thai').default,
  hindiNepaliBengali: require('./dist/scripts/hindi_nepali_bengali').default,
  latin: require('./dist/scripts/arabic').default,
  mundo: require('./dist/scripts/arabic').default,
  sinhala: require('./dist/scripts/sinhala').default,
  tamil: require('./dist/scripts/tamil').default,
};
