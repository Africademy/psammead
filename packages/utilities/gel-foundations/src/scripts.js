/* eslint-disable global-require */
module.exports = {
  latin: require('../dist/scripts/latin-and-cyrillic').default,
  cyrillic: require('../dist/scripts/latin-and-cyrillic').default,
  latinDiacritics: require('../dist/scripts/latin-with-diacritics').default,
  arabic: require('../dist/scripts/arabic').default,
  arabicPashto: require('../dist/scripts/arabic-pashto').default,
  bengali: require('../dist/scripts/bengali').default,
  burmese: require('../dist/scripts/burmese').default,
  hindi: require('../dist/scripts/devanagari-and-gurmukhi').default,
  nepali: require('../dist/scripts/devanagari-and-gurmukhi').default,
  sinhalese: require('../dist/scripts/sinhalese').default,
  tamil: require('../dist/scripts/tamil').default,
  thai: require('../dist/scripts/thai').default,
  chinese: require('../dist/scripts/no-ascenders-or-descenders').default,
  korean: require('../dist/scripts/no-ascenders-or-descenders').default,
  ethiopic: require('../dist/scripts/no-ascenders-or-descenders').default,
};
