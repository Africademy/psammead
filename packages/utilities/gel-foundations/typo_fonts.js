module.exports = require('./dist/typo_fonts'); // eslint-disable-line import/no-unresolved
