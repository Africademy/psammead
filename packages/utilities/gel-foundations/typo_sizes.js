module.exports = require('./dist/typo_sizes'); // eslint-disable-line import/no-unresolved
