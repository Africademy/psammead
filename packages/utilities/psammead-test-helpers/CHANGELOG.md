# Psammead Test Helpers Changelog

| Version | Description |
|---------|-------------|
| 0.1.2   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
