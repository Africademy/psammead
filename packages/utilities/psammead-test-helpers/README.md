# psammead-test-helpers &middot; [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/BBC-News/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/psammead-test-helpers.svg)](https://www.npmjs.com/package/@bbc/psammead-test-helpers) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://reactjs.org/docs/how-to-contribute.html#your-first-pull-request)

This package is a collection helper methods used for snapshot testing, with support for styled components.

## Installation

```jsx
npm install @bbc/psammead-test-helpers --save
```

## Usage

```jsx
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';

shouldMatchSnapshot(
  'should render correctly',
  <h1>Hello World</h1>,
);
```

## Roadmap

There is currently a plan to migrate from `react-test-renderer` to `enzyme` as it provides a nicer higher-level API over the same functionality, as well as having support for more granular unit tests.

## Contributing

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead respository](https://github.com/BBC-News/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/BBC-News/psammead/blob/latest/LICENSE).
