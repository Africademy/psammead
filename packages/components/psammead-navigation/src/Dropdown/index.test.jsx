import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { latin } from '@bbc/gel-foundations/scripts';
import {
  CanonicalDropdown,
  DropdownNavigationLi,
  CanonicalHamburgerMenu,
} from './index';
import pidginNavData from '../../testHelpers/pidgin';

describe('Navigation', () => {
  shouldMatchSnapshot(
    'dropdown should render correctly',
    <CanonicalDropdown announcedText="Menu" closeAction={() => {}}>
      {pidginNavData.map((item, index) => {
        const active = index === 3;
        const { title, url } = item;

        return (
          <DropdownNavigationLi
            script={latin}
            service="news"
            url={url}
            key={title}
            active={active}
            currentPageText="Current page"
          >
            {title}
          </DropdownNavigationLi>
        );
      })}
    </CanonicalDropdown>,
  );
});

describe('Hamburger menu', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <CanonicalHamburgerMenu announcedText="Menu" openAction={() => {}} />,
  );
});
