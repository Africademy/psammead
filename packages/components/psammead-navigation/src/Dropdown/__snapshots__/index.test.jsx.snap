// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger menu canonical should render correctly 1`] = `
.c1 {
  color: #fff;
  fill: currentColor;
}

.c0 {
  width: 2.75rem;
  height: 2.75rem;
  background-color: #B80000;
  padding: 0;
  margin: 0;
  border: 0;
}

.c0:focus {
  box-shadow: inset 0 0 0 0.25rem #FFFFFF;
}

@media (min-width:37.5rem) {
  .c0 {
    display: none;
  }
}

<button
  aria-expanded="false"
  aria-label="Menu"
  class="c0"
>
  <svg
    class="c1"
    height="44px"
    width="44px"
  >
    <path
      d="M12 29h21v-2.333H12V29zm0-5.833h21v-2.334H12v2.334zM12 15v2.333h21V15H12z"
    />
  </svg>
</button>
`;

exports[`Navigation canonical dropdown should render correctly 1`] = `
.c7 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c2 {
  color: #fff;
  fill: currentColor;
}

.c0 {
  background-color: #222222;
}

.c1 {
  background-color: #222222;
  width: 2.75rem;
  height: 2.75rem;
  padding: 0;
  margin: 0;
  border: 0;
}

.c1:focus {
  box-shadow: inset 0 0 0 0.25rem #FFFFFF;
}

.c3 {
  list-style-type: none;
  margin: 0;
  padding-right: 0.5rem;
  padding-left: 0.5rem;
  border-bottom: 0.125rem solid #3F3F42;
}

.c4 {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  border-bottom: 0.0625rem solid #3F3F42;
}

.c4:last-child {
  padding-bottom: 1rem;
  border: 0;
}

.c5 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6 {
  border-left: 0.25rem solid #FFFFFF;
  padding-left: 0.5rem;
}

@media (min-width:37.5rem) {
  .c0 {
    display: none;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c5 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c5 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

<div
  class="c0"
>
  <button
    aria-expanded="true"
    aria-label="Menu"
    class="c1"
  >
    <svg
      class="c2"
      height="44px"
      width="44px"
    >
      <path
        d="M26.741 15L21.6 20.142 16.458 15 15 16.458l5.142 5.142L15 26.742l1.458 1.458 5.142-5.141 5.141 5.141 1.459-1.458-5.142-5.142 5.142-5.142z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <ul
    class="c3"
    role="list"
  >
    <li
      class="c4"
      role="listitem"
    >
      <a
        class="c5"
        href="/pidgin"
      >
        Home
      </a>
    </li>
    <li
      class="c4"
      role="listitem"
    >
      <a
        class="c5"
        href="/pidgin/topics/3d5d5e30-dd50-4041-96d5-c970b20005b9"
      >
        Nigeria
      </a>
    </li>
    <li
      class="c4"
      role="listitem"
    >
      <a
        class="c5"
        href="/pidgin/topics/d2c2ba68-f9ad-4185-a6d1-7f6437256735"
      >
        Africa
      </a>
    </li>
    <li
      class="c4"
      role="listitem"
    >
      <a
        class="c5"
        href="/pidgin/world"
      >
        <span
          class="c6"
          dir="ltr"
          role="text"
        >
          <span
            class="c7"
          >
            Current page
          </span>
          World
        </span>
      </a>
    </li>
    <li
      class="c4"
      role="listitem"
    >
      <a
        class="c5"
        href="/pidgin/media/video"
      >
        Video
      </a>
    </li>
    <li
      class="c4"
      role="listitem"
    >
      <a
        class="c5"
        href="/pidgin/media/audio"
      >
        Audio
      </a>
    </li>
    <li
      class="c4"
      role="listitem"
    >
      <a
        class="c5"
        href="/pidgin/sport"
      >
        Sport
      </a>
    </li>
    <li
      class="c4"
      role="listitem"
    >
      <a
        class="c5"
        href="/pidgin/topics/1c3b60a9-14eb-484b-a750-9f5b1aeaac31"
      >
        Entertainment
      </a>
    </li>
    <li
      class="c4"
      role="listitem"
    >
      <a
        class="c5"
        href="/pidgin/popular/read"
      >
        Most popular
      </a>
    </li>
  </ul>
</div>
`;
